checkov/terraform/context_parsers/base_parser.py

```diff
@@ -68,43 +73,41 @@ def _compute_definition_end_line(self, start_line_num):
                     break
         return end_line_num
 
-    def run(self, tf_file, block):
+    def run(self, tf_file, definition_blocks):
         self.tf_file = tf_file
         self.context = {}
         self.file_lines = self._read_file_lines()
-        self.context = self.enrich_definition_block(block)
-        self.context = self._collect_skip_comments()
+        self.context = self.enrich_definition_block(definition_blocks)
+        self.context = self._collect_skip_comments(definition_blocks)
         return self.context
 
-    @abstractmethod
     def get_block_type(self):
-        raise NotImplementedError()
+        return self.definition_type
+
+    @abstractmethod
+    def get_entity_context_path(self, entity_block):
+        raise NotImplementedError
 
-    def enrich_definition_block(self, block):
+    def _is_block_signature(self, line_tokens, entity_context_path):
+        block_type = self.get_block_type()
+        return all(x in line_tokens for x in [block_type] + entity_context_path)
+
+    def enrich_definition_block(self, definition_blocks):
         """
         Enrich the context of a Terraform block
-        :param block: Terraform block, key-value dictionary
+        :param definition_blocks: Terraform block, key-value dictionary
         :return: Enriched block context
         """
         parsed_file_lines = self._filter_file_lines()
-
-        for i, entity_block in enumerate(block):
-            entity_name, entity_type = self.get_entity_name_and_type(entity_block)
-            if not self.context.get(entity_type):
-                self.context[entity_type] = {}
-            if not self.context.get(entity_type).get(entity_name):
-                self.context[entity_type][entity_name] = {}
+        for i, entity_block in enumerate(definition_blocks):
+            entity_context_path = self.get_entity_context_path(entity_block)
             for line_num, line in parsed_file_lines:
                 line_tokens = [x.replace('"', "") for x in line.split()]
-                if all(x in line_tokens for x in [self.get_block_type(), entity_type, entity_name]):
+                if self._is_block_signature(line_tokens, entity_context_path):
```

Comment:
is that returning a boolean value? name hints it is when it does not.

I know it's not part of the original scope, but can you please add class comments and methods comments to the common classes you have modified?

Reply:
@schosterbarak It's actually returns a boolean value
docs are on the way anyway :)